File: crates/accuracy/fixtures/rust/imports.rs
// Module declarations, imports and qualified paths.
//
// A `use` statement creates a local binding, so a reference resolves to the import line and the
// import line resolves to the definition. See "Import resolution" in the crate README.

mod geometry {
    pub struct Rect {
        pub width: i32,
    }

    pub fn unit() -> Rect { //~ depends: Rect@7
        Rect { width: 1 } //~ depends: Rect@7, width@8
    }
}

use geometry::Rect; //~ depends: geometry@6, Rect@7

fn main() {
    let direct = geometry::unit(); //~ depends: geometry@6, unit@11
    let imported = Rect { width: 2 }; //~ depends: Rect@16, width@8
    let _ = direct.width + imported.width; //~ depends: direct@19, imported@20, width@8
}

File: crates/accuracy/fixtures/typescript/imports.ts
// Import forms and the local names they introduce.
//
// The declarations these names refer to live in other files, which single-file analysis cannot see,
// so an import line has nothing to depend on. What it does is introduce a local name, and every
// reference in this file depends on the import line — the same half of the chain that
// `rust/imports.rs` records for a `use` statement.
//
// `helper as aid` introduces `aid` alone: `helper` is the other module's name for it and is not a
// local name at all. See #269.

import { Widget, helper as aid } from "./widgets";
import type { Shape } from "./shapes";
import Fallback from "./fallback";
import * as everything from "./all";

function build(s: Shape): Widget { //~ depends: Shape@12, Widget@11
    const made = new Widget(); //~ depends: Widget@11
    aid(made); //~ depends: aid@11, made@17
    Fallback.init(); //~ depends: Fallback@13
    everything.run(); //~ depends: everything@14
    return made; //~ depends: made@17
}

const built = build({ sides: 3 } as unknown as Shape); //~ depends: build@16, Shape@12
